src/Sulu/Bundle/AdminBundle/Resources/js/components/Pagination/Pagination.js

```diff
@@ -33,23 +33,20 @@ class Pagination extends React.Component<Props> {
     @action componentDidMount() {
         const {currentPage} = this.props;
 
-        if (!currentPage) {
-            return;
-        }
-
-        this.currentInputValue = this.props.currentPage;
+        this.currentInputValue = currentPage;
+        this.validateAndSubmitInputValue();
     }
 
     @action componentDidUpdate(prevProps: Props) {
-        const {currentPage} = this.props;
+        const {currentPage, totalPages} = this.props;
 
         if (prevProps.currentPage !== currentPage) {
-            if (!currentPage) {
-                this.currentInputValue = 1;
-                return;
-            }
-
             this.currentInputValue = currentPage;
+            this.validateAndSubmitInputValue();
+        }
+
+        if (prevProps.totalPages !== totalPages) {
+            this.validateAndSubmitInputValue();
```

Comment:
not excited to do this here, but i have not found a better place 😕 

doing it directly in the `ListStore` is not possible because we do not want to do this for adapters that use the `InfiniteLoadingStrategy` (eg. the `MediaCardAdapter`).

doing it in the `DefaultLoadingStrategy` is not possible because the loading strategy cannot access the `ListStore`.